src/main/java/mate/academy/service/BookServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.service;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.entity.Book;
+
+public class BookServiceImpl implements BookService {
+    private final BookDao bookDao;
+
+    public BookServiceImpl(BookDao bookDao) {
+        this.bookDao = bookDao;
+    }
+
+    @Override
+    public Book createBook(Book book) {
+        return bookDao.create(book);
+    }
+
+    @Override
+    public Optional<Book> getBookById(int id) {
+        return bookDao.findById(id);
+    }
+
+    @Override
+    public List<Book> findAllBooks() {
+        return bookDao.findAll();
+    }
+
+    @Override
+    public Book updateBook(int id, Book book) {
+        book.setId(id);
+        return bookDao.update(book);
+    }
+
+    @Override
+    public boolean deleteBookById(int id) {
```

Comment:
The `deleteBookById` method should use `Long` for the `id` parameter instead of `int`, as specified in the task requirements.